app/connector/FrontendAuthConnector.scala

```diff
@@ -31,14 +35,43 @@ object GovernmentGatewayEnrolment {
   implicit val fmt = Json.format[GovernmentGatewayEnrolment]
 }
 
+case class InternalUserIdentifier(value: String) extends AnyVal
+
+object InternalUserIdentifier {
+  implicit val reads: Reads[InternalUserIdentifier] = new Reads[InternalUserIdentifier] {
+    override def reads(json: JsValue): JsResult[InternalUserIdentifier] =
+      Try((json \ "internalId").as[String])
+        .map(id => JsSuccess(InternalUserIdentifier(id)))
+        .getOrElse(JsError("Error parsing internal user identifier"))
+  }
+
+  implicit def toString(id: InternalUserIdentifier): String = id.value
```

Comment:
I'll have to check, but since this is a tiny-type which toString is used - I assumed it was the one on the `value` (because I thought that at compile time it replaces the wrapper InternalUserIdentifier with the wrapped type) but you might be right.